app/models/mixins/service_configuration_mixin.rb

```diff
@@ -0,0 +1,18 @@
+module ServiceConfigurationMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :configuration_scripts, :through => :service_resources, :source => :resource, :source_type => 'ConfigurationScript'
+    private :configuration_scripts, :configuration_scripts=
+  end
+
+  def configuration_script
+    configuration_scripts.take
```

Comment:
@mkanoor There is only one configuration_script per service. There's the reason we turned the has_many relation to private and mimic a has_one relation. The has_many relation is inherited from the superclass.